### PR TITLE
Protect against XSS in diff-view by using application/json

### PR DESCRIPTION
**Pull request by olsen232**
## Description

As discussed in https://github.com/koordinates/kart/pull/883
No extra tests yet, this is all the time I was willing to spend on it for right now.

Not clear if tests would be super valuable - we could test the following:
- does html.escape escape the characters that it says it escapes
- does json.dumps create valid json (ie, can we load it using json.loads)
but honestly it seems like these would be pretty well tested already

We can't really test the following:
- is html.escape guaranteed XSS safe for whatever browser the user has
- is json.dumps guaranteed XSS safe for whatever browser the user has (ie, it won't get tripped up by `{"xss": "</script><script>alert(1)</script>"}` or some nonsense)
- does the user's browser respect <script type="application/json"> and not allow XSS inside it? (my one does so that's good)

## Checklist:

- [x] Have you reviewed your own change?
- [ ] Have you included test(s)?
- [x] Have you updated the [changelog](https://github.com/koordinates/kart/blob/master/CHANGELOG.md)?

### CODE REVIEW

**Comment by olsen232** [comment]
@enricofer Find slightly improved diff-view.html template that is more secure against XSS attacks - you can copy the changes to your own HTML template(s).

File: kart/diff-view.html [hunk 1]
```diff
@@ -103,8 +103,9 @@
             top: 2px;
         }
     </style>
-    <script id="kart-data">const DATA=${geojson_data};</script>
+    <script id="kart-data" type="application/json">${geojson_data}</script>
```
Comment (craigds) [inline kart/diff-view.html:106]: what happens if geojson_data contains a string with `</script>` in it - does the nature of this json script tag ignore html in strings like that? or do we need to escape `/` characters or something?

django's `json_script` template filter might be worth checking out: https://docs.djangoproject.com/en/4.2/ref/templates/builtins/#std-templatefilter-json_script

Comment (craigds) [reply kart/diff-view.html:106]:
```html
<!DOCTYPE html>
<html>
    <head>
        <script type="application/json">
            {
                "key": "</script><script>alert('xss')</script><script>{
            }
        </script>
    </head>
</html>
```

this alerts for me. so I think we need to escape `/` or something

Comment (olsen232) [reply kart/diff-view.html:106]: Good catch. Done, and added a test at last

**Review by craigds — APPROVED** [review]

**Pull request merged**